pkg/jobs/canary/canary_jobs.go

```diff
@@ -50,36 +56,36 @@ func StartScanCanaryConfigs(dataFile string, configFiles []string) {
 }
 
 type CanaryJob struct {
-	*kommons.Client
-	Kubernetes kubernetes.Interface
-	Canary     v1.Canary
-	DBCanary   pkg.Canary
-	LogPass    bool
-	LogFail    bool
+	Canary   v1.Canary
+	DBCanary pkg.Canary
+	LogPass  bool
+	LogFail  bool
 }
 
-func (job CanaryJob) GetNamespacedName() types.NamespacedName {
-	return types.NamespacedName{Name: job.Canary.Name, Namespace: job.Canary.Namespace}
+func (j CanaryJob) GetNamespacedName() types.NamespacedName {
+	return types.NamespacedName{Name: j.Canary.Name, Namespace: j.Canary.Namespace}
 }
 
 var minimumTimeBetweenCanaryRuns = 10 * time.Second
 var canaryLastRuntimes = sync.Map{}
 
-func (job CanaryJob) Run() {
-	if runner.IsCanaryIgnored(&job.Canary.ObjectMeta) {
-		return
+func (j CanaryJob) Run(ctx dutyjob.JobRuntime) error {
+	ctx.GetSpan().SetAttributes(attribute.String("canary-id", j.DBCanary.ID.String()))
```

Comment:
I don't think we should be recording traces for canary runs - they are traced internally using check_status, not sure what pushing them to opentelemetry would add ? A slow canary is more likely due to a slow remote server than a slow canary checker db ?

Reply:
Dit it for general observability, like record all transactions to have complete overview and have a general idea of how much time the checks run is taking compared to the other database calls in the job